query-engine/driver-adapters/src/conversion/js_arg_type.rs

```diff
@@ -0,0 +1,91 @@
+#[derive(Debug, PartialEq)]
+pub enum JSArgType {
+    /// 32-bit signed integer.
+    Int32,
+    /// 64-bit signed integer.
+    Int64,
+    /// 32-bit floating point.
+    Float,
+    /// 64-bit floating point.
+    Double,
+    /// String value.
+    Text,
+    /// Database enum value.
+    Enum,
+    /// Database enum array (PostgreSQL specific).
+    EnumArray,
+    /// Bytes value.
+    Bytes,
+    /// Boolean value.
+    Boolean,
+    /// A single character.
+    Char,
+    /// An array value (PostgreSQL).
+    Array,
+    /// A numeric value.
+    Numeric,
+    /// A JSON value.
+    Json,
+    /// A XML value.
+    Xml,
+    /// An UUID value.
+    Uuid,
+    /// A datetime value.
+    DateTime,
+    /// A date value.
+    Date,
+    /// A time value.
+    Time,
+}
+
+impl core::fmt::Display for JSArgType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            JSArgType::Int32 => "Int32",
+            JSArgType::Int64 => "Int64",
+            JSArgType::Float => "Float",
+            JSArgType::Double => "Double",
+            JSArgType::Text => "Text",
+            JSArgType::Enum => "Enum",
+            JSArgType::EnumArray => "EnumArray",
+            JSArgType::Bytes => "Bytes",
+            JSArgType::Boolean => "Boolean",
+            JSArgType::Char => "Char",
+            JSArgType::Array => "Array",
+            JSArgType::Numeric => "Numeric",
+            JSArgType::Json => "Json",
+            JSArgType::Xml => "Xml",
+            JSArgType::Uuid => "Uuid",
+            JSArgType::DateTime => "DateTime",
+            JSArgType::Date => "Date",
+            JSArgType::Time => "Time",
+        };
+
+        write!(f, "{}", s)
+    }
+}
+
+pub fn value_to_js_arg_type(value: &quaint::Value) -> Option<JSArgType> {
+    match &value.typed {
+        quaint::ValueType::Int32(Some(_)) => Some(JSArgType::Int32),
```

Comment:
I think we don't need to unpack internal value and don't need output of this function  to be an `Option`, now, when all quaint options are covered. If the value on this line is `None`, type is still `Int32`.

Reply:
Alright, this required an adjustment in the prisma/prisma repo (https://github.com/prisma/prisma/pull/24922/commits/eef4a1a416e3a0a4857267f60c11ca0d86a9c39f), but it is done.